README.example.md

```diff
@@ -51,32 +53,88 @@ This section should include any business related explanation that helps understa
 If the project is using some special external services (NemID, oAuth, Customer API, etc.) give a short description here.
 Example: The app is using NemID to authenticate all its users.
 
-# Requirements
+# Getting the project running
 
-You must have the following installed and available on your machine:
+There are three supported ways to run the project:
 
-- **Ruby 3.0.x**
-- **Node JS 16.x**
-- **Yarn 1.x**
-- **PostgreSQL 12**
-- **Redis**
+1. [Run locally using docker for database servers](#run-locally-using-docker-for-database-servers)
+2. [Run entirely project locally (for non-docker purists)](#run-entirely-project-locally)
+3. [Run inside docker dev container (for docker purists)](#run-inside-docker-dev-container)
 
-## Installing requirements
+## Run locally using docker for database servers
 
-With [asdf](https://asdf-vm.com/) and [docker](https://www.docker.com/) (or [podman](https://podman.io/)) you can easily install/run the required versions of runtimes and services.
+Requirements:
 
-Install runtimes with `asdf`:
+- docker (or podman)
+- asdf
 
-```
+1. First set up node and ruby:
+
+```bash
+# Add nodejs and ruby plugins if you don't already have them
+asdf plugin add nodejs
+asdf plugin add ruby
+
+# Install ruby and nodejs
 asdf install
+
+# Install yarn
+npm install -g yarn
+```
+
+2. Then run database with docker:
+
+```bash
+docker compose up
 ```
 
-Start services with `docker-compose`:
+3. Run setup:
 
+```bash
+bin/setup
 ```
-docker-compose up
+
+The project is now ready to run locally.
+
+## Run entirely project locally
+
+The steps are the same as for [Run locally using docker for database servers](#run-locally-using-docker-for-database-servers) except for step 2, and `docker` is (obviously) not a requirement.
+
+You will instead need to install the database servers and configure them correctly:
+
+- PostgreSQL (see version in docker-compose.yml)
+- Redis (see version in docker-compose.yml)
+
+It will be necessary to configure the database connections by setting corresponding env vars.
+If postgres is set up to use trust authentication (default for homebrew), then the following should be enough\*:
+
+```bash
+# .env.local
+DATABASE_SERVER=""
 ```
 
+\*_Caution: If you on a mac do not clear `DATABASE_SERVER`, you might get a seemingly unrelated: `+[__NSCFConstantString initialize] may have been in progress in another thread...`_
```

Comment:
Seems odd why it would do a difference, since it defaults to `""`, when the variable is not set.
> host: <%= URI(ENV.fetch("DATABASE_SERVER", '')).host %>

Reply:
This PR adds a default `DATABASE_SERVER` to the template (in `.env`) so that docker will work out of the box.
Several apple silicon mac-devs have faced the mentioned ruby error on other projects where we've done the same. It doesn't matter if the connection string is also correct for the local postgres server.

I added a caution-note to hopefully prevent us from wasting further time on this bug.

When the connection settings are `nil` (and according to @reinisla's testing also when they are `""`) rails will - to my best knowledge - connect to postgres through a socket, whereas it will otherwise use TCP/IP. I think it is this difference that works around the ruby bug.

Reply:
So you need to overwrite the `DATABASE_SERVER` set in `.env`.
In that case, I think we should move `DATABASE_SERVER` from `.env` to `docker-compose.yml`.

Reply:
>  In that case, I think we should move DATABASE_SERVER from .env to docker-compose.yml.

That is a great idea!

Reply:
I realize now that setting the env var in the Docker compose file will not work for the hybrid setup that most of our devs are using. E.g. running DBs in docker, but the rails app locally. Unless I'm missing something here?

Reply:
Right, so we are already setting it in both `.env` (for "hydrid") and overriding it in `docker-compose.yml` (for all docker).
This section says that people using no docker at all, should override the `.env` one with an empty `DATABASE_SERVER`. :+1: 

I have created https://github.com/abtion/rails-template/issues/1941 to discuss this line further.